pkg/api/api.go

```diff
@@ -67,6 +67,145 @@ import (
 // loggerName is the tree path name of the logger for this package.
 const loggerName = "api"
 
+type headerDescriptor struct {
+	// name of header parameter
+	Name string
+
+	// Validate will validate parameter value or
+	// return error if value is not in correct format.
+	Validate func(v string, s *Service) error
+
+	// If this function is defined it will set value to request context
+	SetToContext func(ctx context.Context, parsedValue string) context.Context
+}
+
+var (
+	swarmPinHeader = headerDescriptor{
+		Name: "Swarm-Pin",
+		Validate: func(v string, _ *Service) error {
+			v = strings.ToLower(v)
+			if v == boolHeaderSetValue || v != boolHeaderSetValue || v == "" {
+				return nil
+			}
```

Comment:
`v != boolHeaderSetValue` is not correct

--- 
condition should be:

v == "" || v == "true" || v == "false" 

---


this validate function seam se to repeating, could we reuse it in other places?

Reply:
linter was not letting this repeat func go through but maybe i can write another func for it

Reply:
```
func validateBool(v string, _*Service) error {
 // todo
 return nil
}


...
 headerDescriptor{
	Name: "Swarm-Pin",
	Validate: validateBool,
}
...

```

would this work?